pkg/dist/dist.go

```diff
@@ -35,7 +35,9 @@ func (c *ImageOrURI) DisplayString() string {
 }
 
 type Platform struct {
-	OS string `toml:"os"`
+	OS           string `toml:"os"`
+	Architecture string `toml:"architecture"`
+	Platform     string `toml:"platform,omitempty" json:"platform,omitempty" yaml:"platform,omitempty"`
```

Comment:
This is a bit weird - why is Platform a field on Platform?

Reply:
This Platform struct is used by the `package.toml`, as part of the multi-arch RFC, we are suggesting deprecating this field and adding a set of Targets. 

Maybe we can add a single `Target` as part of this PR? [here](https://github.com/buildpacks/pack/pull/2086/commits/a27ebb3fd627a6669609f9894a76ff912cc56494#diff-66c276bbc91058d8cc79d886cc7ed9d1020973d3cc3620cef91c6695e41451b8) is what I did on the multi-arch PoC.

@jericop what about getting the information you need from there for buildpack packages and extensions?